### PR TITLE
Deleting the Default Queue

### PR DESCRIPTION
**Pull request by aydreeihn**
This commit fixes an issue where Agents were able to delete their default queue which then prevented them from going to any queue upon logging in. Now, they will see an error message if they try to do so.

### CODE REVIEW

File: scp/queues.php [hunk 1]
```diff
@@ -72,11 +73,13 @@
                 if ($queue->save()) $updated++;
                 break;
             case 'delete':
-                if ($queue->delete()) $updated++;
+                if ($queue->getId() == $cfg->get('default_ticket_queue'))
```
Comment (protich) [inline scp/queues.php]: Please use getter `$cfg->getDefaultTicketQueueId()`. Also you don't need to make `$cfg` global.

**Comment by aydreeihn** [comment]
Changes made.

**Comment by protich** [comment]
Please add the diff below to above commit as a fall-back for users who might have deleted the default queue already.

```diff

diff --git a/scp/tickets.php b/scp/tickets.php
index 86fec81..f091841 100644
--- a/scp/tickets.php
+++ b/scp/tickets.php
@@ -23,6 +23,11 @@ require_once(INCLUDE_DIR.'class.json.php');
 require_once(INCLUDE_DIR.'class.dynamic_forms.php');
 require_once(INCLUDE_DIR.'class.export.php');       // For paper sizes
 
+
+
+// Fetch ticket queues organized by root and sub-queues
+$queues = CustomQueue::getHierarchicalQueues($thisstaff);
+
 $page='';
 $ticket = $user = null; //clean start.
 $redirect = false;
@@ -113,18 +118,21 @@ if (!$ticket) {
         $queue = AdhocSearch::load($key);
     }
 
-    // Make the current queue sticky
-    $_SESSION[$queue_key] = $queue_id;
-
-    if ((int) $queue_id && !$queue) {
+    if ((int) $queue_id && !$queue)
         $queue = SavedQueue::lookup($queue_id);
-    }
-    if (!$queue) {
-        $queue = SavedQueue::lookup($cfg->getDefaultTicketQueueId());
-    }
 
-    // Set the queue_id for navigation to turn a top-level item bold
-    $_REQUEST['queue'] = $queue->getId();
+    if (!$queue && ($qid=$cfg->getDefaultTicketQueueId()))
+        $queue = SavedQueue::lookup($qid);
+
+    if (!$queue && $queues)
+        list($queue,) = $queues[0];
+
+    if ($queue) {
+        // Set the queue_id for navigation to turn a top-level item bold
+        $_REQUEST['queue'] = $queue->getId();
+        // Make the current queue sticky
+         $_SESSION[$queue_key] = $queue->getId();
+    }
 }
 
 // Configure form for file uploads
@@ -450,9 +458,6 @@ if (isset($_GET['clear_filter']))
 $nav->setTabActive('tickets');
 $nav->addSubNavInfo('jb-overflowmenu', 'customQ_nav');
 
-// Fetch ticket queues organized by root and sub-queues
-$queues = CustomQueue::getHierarchicalQueues($thisstaff);
-
 // Start with all the top-level (container) queues
 foreach ($queues as $_) {
     list($q, $children) = $_;
```

**Comment by aydreeihn** [comment]
Done

**Pull request merged**